synapse/lib/cell.py

```diff
@@ -1456,54 +1453,78 @@ async def initServiceNetwork(self):
             await self.addHttpsPort(port)
             logger.info(f'https listening: {port}')
 
+    async def getAhaInfo(self):
+
+        # Default to static information
+        ahainfo = self.conf.get('aha:svcinfo')
+        if ahainfo is not None:
+            return ahainfo
+
+        # If we have not setup our dmon listener yet, do not generate ahainfo
+        if self.sockaddr is None:
+            return None
+
+        # If we have no dmon listener configured, do not generate ahainfo
+        turl = self.conf.get('dmon:listen')
+        if turl is None:
```

Comment:
is it possible for this to be none if `self.sockaddr` is set?

Reply:
not with the default initServiceNetwork implementation. We should be safe to remove this early return.